client/engine/engine.go

```diff
@@ -200,6 +212,35 @@ func (e *Engine) handleMessage(message protocols.Message) (ObjectiveChangeEvent,
 		}
 
 		allCompleted.CompletedObjectives = append(allCompleted.CompletedObjectives, progressEvent.CompletedObjectives...)
+
+		// We record any objective ids for proposals still in the queue
+		// Now that we've cranked our objective they may be able to progress
+		for _, rel := range updatedObjective.Related() {
+			c, ok := rel.(*consensus_channel.ConsensusChannel)
+			if ok {
+				for _, p := range c.ProposalQueue() {
+					id := getProposalObjectiveId(p.Proposal)
+
+					toProcessAfter[id] = true
+				}
+			}
+		}
+	}
+
+	// We now check try to process any of the objectives that we may have unblocked
+	for id := range toProcessAfter {
+		objective, err := e.store.GetObjectiveById(id)
+
+		if err != nil {
+			return ObjectiveChangeEvent{}, err
+		}
+
+		progressEvent, err := e.attemptProgress(objective)
```

Comment:
This might not be sufficient. What if the call to `attemptProgress` here means that another proposal is unblocked?

Reply:
Actually I think we should be OK. Since we're adding the objective Ids for **all** remaining proposals in the `ProposalQueue` we would expect the unblocked objective to already be part of `toCompleted`